### ISSUE TITLE
[BUG] Timeout on windows, python 3.7

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#2448]
We seem to be observing an increasing number of timeouts on python version 3.7, windows.

One here:
https://github.com/alan-turing-institute/sktime/pull/2411

another one here (one of the runs):
https://github.com/alan-turing-institute/sktime/pull/2410
https://github.com/alan-turing-institute/sktime/actions/runs/2150278613/

FYI @lmmentel, k1m190r

**Comment by fkiraly** [comment]
@miraep8, unfortunately it is not an isolated issue 😱 

the logs of your first run seem to have disappeared, do you remember anything from them?

**Comment by fkiraly** [comment]
on #2411, last life sign of `gw1`:

```
 utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X1] 
[gw1] [ 95%] PASSED utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X1] 
utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X2] 
[gw1] [ 95%] PASSED utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X2] 
utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X3] 
[gw1] [ 95%] PASSED utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X3] 
utils/validation/tests/test_panel.py::test_check_enforce_min_instances 
[gw1] [ 95%] PASSED utils/validation/tests/test_panel.py::test_check_enforce_min_instances 
utils/validation/tests/test_panel.py::test_check_X_enforce_univariate 
[gw1] [ 95%] PASSED utils/validation/tests/test_panel.py::test_check_X_enforce_univariate 
utils/validation/tests/test_panel.py::test_check_X_enforce_min_columns 
[gw1] [ 95%] PASSED utils/validation/tests/test_panel.py::test_check_X_enforce_min_columns 
Error: The operation was canceled.
```

last life sign of `gw0`: 
```
2022-04-08T10:41:12.9206873Z [gw1] [ 57%] PASSED forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_predict_time_index_in_sample_full[AutoEnsembleForecaster-int-int-True] 
2022-04-08T10:41:12.9207627Z forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_predict_time_index_in_sample_full[AutoEnsembleForecaster-int-int-False] 
2022-04-08T10:41:12.9208374Z [gw1] [ 57%] PASSED forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_predict_time_index_in_sample_full[AutoEnsembleForecaster-int-int-False] 
2022-04-08T10:41:12.9209127Z forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_predict_time_index_in_sample_full[AutoEnsembleForecaster-range-int-True] 
2022-04-08T10:41:12.9209825Z [gw0] [ 57%] PASSED forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_update_predict_single[Prophet-update_params=True-fh=1] 
2022-04-08T10:41:12.9210484Z forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_update_predict_single[Prophet-update_params=True-fh=[2 5]] 
2022-04-08T10:41:12.9211196Z [gw1] [ 57%] PASSED forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_predict_time_index_in_sample_full[AutoEnsembleForecaster-range-int-True] 
2022-04-08T10:41:12.9211951Z forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_predict_time_index_in_sample_full[AutoEnsembleForecaster-range-int-False] 
2022-04-08T10:41:12.9212754Z [gw1] [ 57%] PASSED forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_predict_time_index_in_sample_full[AutoEnsembleForecaster-range-int-False] 
2022-04-08T10:41:12.9213587Z forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_predict_time_index_in_sample_full[AutoEnsembleForecaster-period-int-True] 
2022-04-08T10:41:12.9214729Z [gw0] [ 57%] PASSED forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_update_predict_single[Prophet-update_params=True-fh=[2 5]] 
2022-04-08T10:41:12.9215396Z forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_update_predict_single[Prophet-update_params=False-fh=1] 
```

that´s ... `Prophet` ... why am I not surprised?

**Comment by lmmentel** [comment]
Yup, there's not much in the logs. My guess would be that one process hangs and hogs a lot of the resources and that slows down the other process. Might be worth to add a timeout https://github.com/pytest-dev/pytest-timeout but that will help us debug and not solve the issue. it seems to stalls on one of the _integration_ tests. As an alternative someone with a win machine could try to reproduce with docker and limiting the resources available to container.

**Comment by fkiraly** [comment]
yes, we should at least capture what test it was working on when it timed out - do you know how to add that? Simply adding the plugin and then the `--timeout=600` or so param?

**Comment by miraep8** [comment]
Yep! I do!  Its actually interesting - I am pretty sure that my code also failed at test_check_X_enforce_min columns. Will see if I can go grab the exact message

**Comment by miraep8** [comment]
Here is the last few things that ran in the logs.  Perhaps it is just a coincidence that our tests both reached 6hrs at the same test, perhaps it is the culprit. 

`
[gw1] [ 93%] PASSED utils/validation/tests/test_check_window_length.py::test_window_length_bad_arg[-2.4-10] 
utils/validation/tests/test_check_window_length.py::test_window_length_bad_arg[0.2-None] 
[gw1] [ 93%] PASSED utils/validation/tests/test_check_window_length.py::test_window_length_bad_arg[0.2-None] 
utils/validation/tests/test_forecasting.py::test_check_fh_empty_input[arg0] 
[gw1] [ 93%] PASSED utils/validation/tests/test_forecasting.py::test_check_fh_empty_input[arg0] 
utils/validation/tests/test_forecasting.py::test_check_fh_empty_input[arg1] 
[gw1] [ 93%] PASSED utils/validation/tests/test_forecasting.py::test_check_fh_empty_input[arg1] 
utils/validation/tests/test_forecasting.py::test_check_fh_empty_input[arg2] 
[gw1] [ 93%] PASSED utils/validation/tests/test_forecasting.py::test_check_fh_empty_input[arg2] 
utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X0] 
[gw1] [ 93%] PASSED utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X0] 
utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X1] 
[gw1] [ 93%] PASSED utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X1] 
utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X2] 
[gw1] [ 93%] PASSED utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X2] 
utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X3] 
[gw1] [ 93%] PASSED utils/validation/tests/test_panel.py::test_check_X_bad_input_args[X3] 
utils/validation/tests/test_panel.py::test_check_enforce_min_instances 
[gw1] [ 93%] PASSED utils/validation/tests/test_panel.py::test_check_enforce_min_instances 
utils/validation/tests/test_panel.py::test_check_X_enforce_univariate 
[gw1] [ 93%] PASSED utils/validation/tests/test_panel.py::test_check_X_enforce_univariate 
utils/validation/tests/test_panel.py::test_check_X_enforce_min_columns 
[gw1] [ 93%] PASSED utils/validation/tests/test_panel.py::test_check_X_enforce_min_columns 
Error: The operation was canceled.
`

**Comment by miraep8** [comment]
An idea -  instead of time timeout we could simply print the 5 slowest tests?  I believe this can be done with the [--duration argument to pytest](https://stackoverflow.com/questions/27884404/printing-test-execution-times-and-pinning-down-slow-tests-with-py-test).

**Comment by miraep8** [comment]
Also - just fyi - I think if you click on the little red x next to the failed test then you are still able to access the details of the failed test! :)  That's how I got the logs!
![little_red_x](https://user-images.githubusercontent.com/10511777/162847194-3f32d00e-e6d1-4eea-af8e-c57d0d356d2f.png)

**Comment by fkiraly** [comment]
> Also - just fyi - I think if you click on the little red x next to the failed test then you are still able to access the details of the failed test! :)

Ah!!! Thanks for this learning moment. Did not know that clicking this got you somewhere.

**Comment by miraep8** [comment]
Ah!  Makes a lot of sense!  Glad we have a resolution!  :)  Makes a lot more sense as I also was unsure why `test_check_X_enforce_min_columns` would cause this issue!

**Comment by miraep8** [comment]
>I thought this was already printed? I always look at the end and the list is there, e.g., here: https://github.com/alan-turing-institute/sktime/issues/2377

Ah yes, I see now that the pytest is configured to print the time of each test on a successful test! :)  Would be nice if there was someway to access this information in the event of a timeout as well.  - I just compared both and it seems to be missing at the bottom of the tests which timed out.

**Comment by fkiraly** [comment]
So, digging into the logs more:

as far as I now understand, each worker makes status updates in two steps: if a test is passed, it prints the `PASSED` message, and then it prints immediately the next test it starts to run.

So, in the log above, it would mean that `gw0` started `test_update_predict_single[Prophet-update_params=False-fh=1]` and got stuck there.
